### ISSUE TITLE
"deleted object would be resaved by cascade"-Exception when reading uncached ACL (not in HibernateCache but dedicated aclCache) within same HibernateSession

### ISSUE THREAD

**Issue by dariusan** [grails-plugins/grails-spring-security-acl#25]
Execute the following code in the Grails web console to reproduce the exception

```
DomainO.withTransaction {
        DomainO anyDomainOWithAclEntry = DomainO.first() //we assume there exists an ACL entry for anyDomainOWithAclEntry
        ctx.aclCache.clearCache() // the problem only occurs if the ACL is not yet cached. We therefore clear the cache first to make sure it happens
        println ctx.aclUtilService.readAcl(anyDomainOWithAclEntry)
        ctx.aclUtilService.deleteAcl(anyDomainOWithAclEntry)
}
```

Workaround or fix:

edit class AclService.deleteEntries() accordingly:

```
//replace the following line
AclEntry.where { aclObjectIdentity == oid }.deleteAll()
//with 
AclEntry.findAllByAclObjectIdentity(oid)*.delete()
//which does not use DML statements and updates first level cache correctly

```

**Comment by jameskleeh** [comment]
@dariusan Can you upload an example project that demonstrates the issue?

**Comment by jameskleeh** [comment]
Fixed in 3.1.0
